### PR TITLE
[Playground] Upgrade Flutter linter, fix linter issues (#21951)

### LINKED ISSUE #21951: [Playground Task]: Upgrade Flutter linter
### What needs to happen?

Flutter 3 comes with new lints. For instance, they could have saved us from this bug that had been affecting analytics reporting: https://github.com/apache/beam/pull/21926#discussion_r900370313

We should examine all new lints and enable those making sense to the project.

### Issue Priority

Priority: 3

### Issue Component

Component: beam-playground

### PR DESCRIPTION
**Pull request by alexeyinkin**
Resolves #21951 

------------------------

Thank you for your contribution! Follow this checklist to help us incorporate your contribution quickly and easily:

 - [x] [**Choose reviewer(s)**](https://beam.apache.org/contribute/#make-your-change) and mention them in a comment (`R: @username`).
 - [x] Mention the appropriate issue in your description (for example: `addresses #123`), if applicable. This will automatically add a link to the pull request in the issue. If you would like the issue to automatically close on merging the pull request, comment `fixes #<ISSUE NUMBER>` instead.
 - [ ] Update `CHANGES.md` with noteworthy changes.
 - [x] If this contribution is large, please file an Apache [Individual Contributor License Agreement](https://www.apache.org/licenses/icla.pdf).

See the [Contributor Guide](https://beam.apache.org/contribute) for more tips on [how to make review process smoother](https://beam.apache.org/contribute/#make-reviewers-job-easier).

To check the build health, please visit [https://github.com/apache/beam/blob/master/.test-infra/BUILD_STATUS.md](https://github.com/apache/beam/blob/master/.test-infra/BUILD_STATUS.md)

GitHub Actions Tests Status (on master branch)
------------------------------------------------------------------------------------------------
[Build python source distribution and wheels](https://github.com/apache/beam/actions?query=workflow%3A%22Build+python+source+distribution+and+wheels%22+branch%3Amaster+event%3Aschedule)
[Python tests](https://github.com/apache/beam/actions?query=workflow%3A%22Python+Tests%22+branch%3Amaster+event%3Aschedule)
[Java tests](https://github.com/apache/beam/actions?query=workflow%3A%22Java+Tests%22+branch%3Amaster+event%3Aschedule)

See [CI.md](https://github.com/apache/beam/blob/master/CI.md) for more information about GitHub Actions CI.

### CODE REVIEW

File: playground/frontend/analysis_options.yaml [hunk 1]
```diff
@@ -28,7 +28,12 @@ include: package:flutter_lints/flutter.yaml
 
 # exclude generated files
 analyzer:
-  exclude: [lib/api/**, test/**.mocks.dart, lib/**.g.dart, lib/l10n/**.arb, build/**]
```
Comment (alexeyinkin) [inline playground/frontend/analysis_options.yaml:31]: So we can easily edit the lines individually.

File: playground/frontend/lib/modules/examples/components/example_list/category_expansion_panel.dart [hunk 2]
```diff
@@ -54,8 +54,8 @@ class CategoryExpansionPanel extends StatelessWidget {
             children: [
               Transform.rotate(
                 angle: -math.pi * animationValue,
-                child: const Icon(Icons.keyboard_arrow_down),
```
Comment (alexeyinkin) [inline playground/frontend/lib/modules/examples/components/example_list/category_expansion_panel.dart:57]: Linter now recommends to order 'child' property last as it tends to be the largest one.

File: playground/frontend/lib/modules/examples/example_selector.dart [hunk 3]
```diff
@@ -153,7 +153,7 @@ class _ExampleSelectorState extends State<ExampleSelector>
                       child: SlideTransition(
                         position: offsetAnimation,
                         child: Material(
-                          elevation: kElevation.toDouble(),
+                          elevation: kElevation,
```
Comment (alexeyinkin) [inline playground/frontend/lib/modules/examples/example_selector.dart:156]: It is already double, linter now flags that.

File: playground/frontend/lib/pages/playground/states/examples_state.dart [hunk 4]
```diff
@@ -50,31 +50,31 @@ class ExampleState with ChangeNotifier {
   }
 
   Future<String> getExampleOutput(String id, SDK sdk) async {
-    return await _exampleRepository.getExampleOutput(
+    return _exampleRepository.getExampleOutput(
```
Comment (alexeyinkin) [inline playground/frontend/lib/pages/playground/states/examples_state.dart:53]: Returning the future is the same as awaiting and then returning. Linter now flags this.

File: playground/frontend/pubspec.yaml [hunk 5]
```diff
@@ -35,7 +35,7 @@ publish_to: 'none' # Remove this line if you wish to publish to pub.dev
 version: 1.0.0+1
 
 environment:
-  sdk: ">=2.12.0 <3.0.0"
+  sdk: ">=2.17.0 <3.0.0"
```
Comment (alexeyinkin) [inline playground/frontend/pubspec.yaml:38]: We have already upgraded to Flutter 3. It requires Dart 2.17+ so it was used anyway.

File: playground/frontend/pubspec.yaml [hunk 6]
```diff
@@ -44,37 +44,32 @@ environment:
 # the latest version available on pub.dev. To see which dependencies have newer
 # versions available, run `flutter pub outdated`.
 dependencies:
```
Comment (alexeyinkin) [inline playground/frontend/pubspec.yaml:46]: Linter requires alphabetic ordering of dependencies.

File: playground/frontend/pubspec.yaml [hunk 7]
```diff
@@ -44,37 +44,32 @@ environment:
...
 dev_dependencies:
+  build_runner: ^2.1.4
+  flutter_lints: ^2.0.1
   flutter_test:
     sdk: flutter
   mockito: ^5.0.16
-  build_runner: ^2.1.4
-  google_fonts: ^2.3.1
-  usage: ^4.0.2
-  collection: ^1.15.0
```
Comment (alexeyinkin) [inline playground/frontend/pubspec.yaml:70]: These three are normal dependencies and not dev-time ones.

**Review by damondouglas — APPROVED** [review]
@pabloem LGTM

**Comment by pabloem** [comment]
lgtm thanks!

**Pull request merged**